### PR TITLE
update to NSB v7

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

**Comment by SimonCropp** [comment]
@Particular/nservicebus-maintainers @danielmarbach 

something seems to  be wrong with the async context flow

specifically when i debug it this line is hit https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.AcceptanceTesting/ScenarioWithContext.cs#L37

But when it gets to here https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.AcceptanceTesting/ContextAppender.cs#L9 then `ScenarioContext.Current` returns null. 

this results in the below exception.

```
System.NullReferenceException occurred
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=NServiceBus.AcceptanceTesting
  StackTrace:
   at NServiceBus.AcceptanceTesting.ContextAppender.get_IsDebugEnabled()
```

Any help would be appreciated

**Comment by andreasohlund** [comment]
We have the same issue in ASB, @timbussman is working on a fix/workaround

File: src/NServiceBus.Wcf.AcceptanceTests/DefaultServer.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        recoverability.Immediate(immediate => immediate.NumberOfRetries(0));
+        var storageDir = Path.Combine(NServiceBusAcceptanceTest.StorageRootDir, NUnit.Framework.TestContext.CurrentContext.Test.ID);
+
+        configuration.UseTransport<LearningTransport>()
+            .StorageDirectory(storageDir);
+
+        configuration.RegisterComponentsAndInheritanceHierarchy(runDescriptor);
+
+
+        configuration.GetSettings().SetDefault("ScaleOut.UseSingleBrokerQueue", true);
```
Comment (timbussmann) [inline src/NServiceBus.Wcf.AcceptanceTests/DefaultServer.cs]: I don't think that setting is used anymore?

File: src/NServiceBus.Wcf.AcceptanceTests/EndpointCustomizationConfigurationExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        var assemblies = new AssemblyScanner().GetScannableAssemblies();
+
+        var types = assemblies.Assemblies
+            //exclude all test types by default
+            .Where(a =>
+            {
+                var references = a.GetReferencedAssemblies();
+
+                return references.All(an => an.Name != "nunit.framework");
```
Comment (timbussmann) [inline src/NServiceBus.Wcf.AcceptanceTests/EndpointCustomizationConfigurationExtensions.cs]: that should no longer be filtered that way as it excludes the testing library now (as it references Nunit).

**Comment by andreasohlund** [comment]
@SimonCropp pushed a fix for the null ref when logging. Also fixed a few other tests now that callbacks needs to be explicitly enabled. Still 2 failing tests which seems to be related to them using `NetNamedPipeBinding`.

**Comment by andreasohlund** [comment]
@timbussmann your comments have been addressed

**Comment by SimonCropp** [comment]
@andreasohlund the setup on the WcfSupport feature is not firing. any ideas why?

**Comment by andreasohlund** [comment]
Got to be assembly scanning issue?

**Pull request merged**